### ISSUE TITLE
Translatable Weathercode Description Field

### ISSUE THREAD

**Issue by ghost** [open-meteo/open-meteo#287]
As suggested in #262 and #228 it would be nice to have a weathercode description that could be translated into different languages and using a lang parameter people get the description in ther language. It could be added when someone chooses to include the weathercode. Here's a suggestion for what could be used for English:

Code | Description
-- | --
0 | Clear
1 | Mostly Clear
2 | Partly Cloudy
3 | Cloudy
45 | Fog
48 | Freezing Fog
51 | Light Drizzle
53 | Drizzle
55 | Heavy Drizzle
56 | Light Freezing Drizzle
57 | Freezing Drizzle
61 | Light Rain
63 | Rain
65 | Heavy Rain
66 | Light Freezing Rain
67 | Freezing Rain
71 | Light Snow
73 | Snow
75 | Heavy Snow
77 | Snow Grains
80 | Light Rain Shower
81 | Rain Shower
82 | Heavy Rain Shower
85 | Snow Shower
86 | Heavy Snow Shower
95 | Thunderstorm
96 | Hailstorm
99 | Heavy Hailstorm

Is weathercode 96 and 99 thunderstorms with hail or just stronger thunderstorms? The docs say it's supposed to be hail but I don't think the code calculation reflects that or maybe it does and I don't know how to read the code properly.

**Comment by long-nguyen** [comment]
Hi, I want to ask why your WMO code is lacking from global list?
https://www.nodc.noaa.gov/archive/arc0021/0002199/1.1/data/0-data/HTML/WMO-CODE/WMO4677.HTM

**Comment by patrick-zippenfenig** [comment]
Yes, dust and other aerosols like smoke are available in global air pollution models. Some are available in the [Air Quality API](https://open-meteo.com/en/docs/air-quality-api) others are available from the [CAMS global atmospheric composition forecasts](https://ads.atmosphere.copernicus.eu/cdsapp#!/dataset/cams-global-atmospheric-composition-forecasts?tab=overview).

Adding dust/smoke/haze is a good idea, but can lead to inconsistencies as many different models are combined. It also raises issues like breaking changes to the API as new codes my break existing implementations from other users. Once option could be a `weathercodeV2` with an extended set that is also more optimised for DMO instead of manned observations.

**Comment by mirix** [comment]
I have created a JSON file mapping WMO code 4677 to:

1. A short description in English.

2. Animated SVG icons (https://basmilius.github.io/weather-icons/index-line.html)

The Open-Meteo-powered app is up and running here (beta):

https://huggingface.co/spaces/mirix/weather_for_hikers

You can check the code and the mapping.

**Comment by Benjamin-Loison** [comment]
Based on https://open-meteo.com/en/docs#weather_variable_documentation:

<details>
<summary>JSON:</summary>

```json
{
    "0": "Clear sky",
    "1": "Mainly clear",
    "2": "Partly cloudy",
    "3": "Overcast",
    "45": "Fog",
    "48": "Depositing rime fog",
    "51": "Light drizzle",
    "53": "Moderate drizzle",
    "55": "Dense drizzle",
    "56": "Light freezing drizzle",
    "57": "Dense freezing drizzle",
    "61": "Slight rain",
    "63": "Moderate rain",
    "65": "Heavy rain",
    "66": "Light freezing rain",
    "67": "Heavy freezing rain",
    "71": "Slight snow fall",
    "73": "Moderate snow fall",
    "75": "Heavy snow fall",
    "77": "Snow grains",
    "80": "Slight rain shower",
    "81": "Moderate rain shower",
    "82": "Violent rain shower",
    "85": "Slight snow shower",
    "86": "Heavy snow shower",
    "95": "Thunderstorm",
    "96": "Slight hail thunderstorm",
    "99": "Heavy hail thunderstorm"
}
```
</details>

Personal notes: [Benjamin_Loison/Voice_assistant/issues/33#issuecomment-11386192](https://codeberg.org/Benjamin_Loison/Voice_assistant/issues/33#issuecomment-11386192)

**Comment by terraputix** [comment]
Hi,
we improved the documentation of the weather code table a bit, compare https://github.com/open-meteo/open-meteo-website/pull/1171.
As the `weather_code` is a just a _code_, no additional translations/localization will be added.
